ckan/lib/redis.py

```diff
@@ -0,0 +1,63 @@
+# encoding: utf-8
+
+u'''
+Redis utilities.
+
+.. versionadded:: 2.6
+'''
+
+from __future__ import absolute_import
+
+import datetime
+import logging
+
+from pylons import config
+from redis import ConnectionPool, Redis
+
+
+log = logging.getLogger(__name__)
+
+REDIS_URL_SETTING_NAME = u'ckan.redis.url'
+
+REDIS_URL_DEFAULT_VALUE = u'redis://localhost:6379/0'
```

Comment:
We've added a hard failure for when site_url isn't configured now instead of having a site up but sometimes generating broken links. Most of the logic functions that return datasets now depend on package_search (and solr) working so I'd argue we should fail in that case too.

This is a bit different. As written the code assumes a local redis install and starts using it. That comes with a (small) chance of corrupting data from another program. Not a nice thing to do. My vote is for hard failure, what do the rest of you think?